src/main/java/com/yelp/nrtsearch/server/module/S3Module.java

```diff
@@ -82,11 +83,14 @@ protected AmazonS3 providesAmazonS3(LuceneServerConfiguration luceneServerConfig
         clientBuilder.setClientConfiguration(clientConfiguration);
       }
 
-      return clientBuilder
-          .withCredentials(awsCredentialsProvider)
-          .withEndpointConfiguration(
-              new AwsClientBuilder.EndpointConfiguration(serviceEndpoint, region))
-          .build();
+      AmazonS3ClientBuilder amazonS3ClientBuilder =
```

Comment:
The `clientBuilder` is already a local reference to the `AmazonS3ClientBuilder`. Do we need to create another one?